mlrun/serving/server.py

```diff
@@ -633,12 +667,21 @@ async def async_execute_graph(
     if server.verbose:
         context.logger.info(server.to_yaml())
 
-    df = data.as_df()
-
     responses = []
 
     async def run(body):
         event = storey.Event(id=index, body=body)
+        if timestamp_column:
+            body = event.body
+            if not isinstance(body, dict):
+                raise mlrun.errors.MLRunRuntimeError(
+                    f"When timestamp_column=True, event body must be a dict – got {type(body).__name__} instead"
+                )
+            if timestamp_column not in body:
+                raise mlrun.errors.MLRunRuntimeError(
+                    f"Event body '{body}' did not contain timestamp column '{timestamp_column}'"
+                )
+            event._original_timestamp = body[timestamp_column]
```

Comment:
What will happen if you have a batch? In this case the `body[timestamp_column]` is a list.